examples/platform/telink/app/src/Service.cpp

```diff
@@ -0,0 +1,96 @@
+/*
+ *
+ *    Copyright (c) 2021 Project CHIP Authors
+ *
+ *    Licensed under the Apache License, Version 2.0 (the "License");
+ *    you may not use this file except in compliance with the License.
+ *    You may obtain a copy of the License at
+ *
+ *        http://www.apache.org/licenses/LICENSE-2.0
+ *
+ *    Unless required by applicable law or agreed to in writing, software
+ *    distributed under the License is distributed on an "AS IS" BASIS,
+ *    WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ *    See the License for the specific language governing permissions and
+ *    limitations under the License.
+ */
+
+/**
+ * @file
+ *   This file implements the service publishing code for example usage.
+ */
+
+#include "Service.h"
+
+#include <string.h>
+
+#include <logging/log.h>
+#include <platform/CHIPDeviceLayer.h>
+#include <zephyr.h>
+
+#if CHIP_ENABLE_OPENTHREAD
+#include <openthread/message.h>
+#include <openthread/udp.h>
+#include <platform/OpenThread/OpenThreadUtils.h>
+#include <platform/ThreadStackManager.h>
+#include <platform/telink/ThreadStackManagerImpl.h>
+#endif
+
+LOG_MODULE_DECLARE(app);
+
+#define DEVICE_NAME_MAX 15
+
+// Transport Callbacks
+namespace {
+
+char deviceName[DEVICE_NAME_MAX];
+constexpr uint16_t kUDPBroadcastPort = 23367;
+
+} // namespace
+
+void PublishService()
```

Comment:
I don't think this functionality is needed now that we have OpenThread SRP support in Matter. It should probably be removed from other examples, too.